spec/components/jobseekers/search_results/polygon_visualizer_component_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "rails_helper"
+
+RSpec.describe Jobseekers::SearchResults::PolygonVisualizerComponent, type: :component do
+  subject { described_class.new(vacancies_search: vacancies_search, auth: auth) }
+
+  before do
+    allow(ENV).to receive(:fetch).with("GOOGLE_MAPS_API_KEY").and_return("123secret")
+    allow(vacancies_search).to receive_message_chain(:location_search, :polygon_boundaries).and_return(polygon_boundaries)
+    render_inline(subject)
+  end
+
+  let(:auth) { true }
+  let(:vacancies_search) { instance_double(Search::SearchBuilder) }
+  let(:polygon_boundaries) { [[1, 2, 1, 2], [3, 4, 3, 4]] }
+
+  describe "#render?" do
+    context "when unauthenticated" do
+      context "when auth is not exactly true but is truthy" do
```

Comment:
I don't think this is worth testing tbh – would it ever be not a boolean, and if so, do we really care? See comment about just doing

```ruby
def render?
  @auth
end
```